File: code/client/src/app/card/effect/charge-effect.ts
import { AbstractEffect } from './abstract-effect';

export class ChargeEffect extends AbstractEffect {

    public ChargeEffect() {
        this.name = 'ChargeEffect';
    }

    public applyPreActionEffect(cardUniqueId: string, clientId: number, targetCardUniqueId: string, targetCardEffectName: string): boolean {
        console.log('ChargeEffect -> Applying applyInActionPreCondEffect() | return true');
        return true;
    }
}

File: code/server/src/main/java/minihearthstone/entities/cards/effects/ChargeEffect.java
package minihearthstone.entities.cards.effects;

import minihearthstone.entities.cards.Card;
import minihearthstone.entities.gameboard.GameBoard;

import javax.persistence.Entity;

@Entity
public class ChargeEffect extends AbstractEffect {

    public ChargeEffect() {
        setName("ChargeEffect");
    }

    @Override
    public boolean applyOnChoiceEffect(Card card, Long playerId, GameBoard gameBoard) {
        log.info("ChargeEffect -> Applying applyOnChoiceEffect()");
        card.setCanAttack(true);
        return true;
    }

    @Override
    public boolean applyPreActionEffect(Card card, Long playerId, String targetCardUniqueId, GameBoard gameBoard) {
        log.info("ChargeEffect -> Applying applyInActionPreCondEffect() | return true");
        return true;
    }

    @Override
    public boolean applyInActionPreCondEffect(Card card, Long playerId, GameBoard gameBoard) {
        return true;
    }

    @Override
    public boolean applyInActionPostCondEffect(Card card, Long playerId, GameBoard gameBoard) {
        return true;
    }



}